Libraries/Lists/__tests__/VirtualizedList-test.js

```diff
@@ -456,4 +456,52 @@ describe('VirtualizedList', () => {
       console.error.mockRestore();
     }
   });
+
+  it('throws if using scrollToIndex with index less than 0', () => {
+    const component = ReactTestRenderer.create(
+      <VirtualizedList
+        data={[{key: 'i1'}, {key: 'i2'}, {key: 'i3'}]}
+        renderItem={({item}) => <item value={item.key} />}
+        getItem={(data, index) => data[index]}
+        getItemCount={data => data.length}
+      />,
+    );
+    const instance = component.getInstance();
+
+    expect(() => instance.scrollToIndex({index: -1})).toThrow(
+      'scrollToIndex out of range: requested index -1 but minimum is 0',
+    );
+  });
+
+  it('throws if using scrollToIndex when item length is less than 1', () => {
+    const component = ReactTestRenderer.create(
+      <VirtualizedList
+        data={[]}
+        renderItem={({item}) => <item value={item.key} />}
+        getItem={(data, index) => data[index]}
+        getItemCount={data => data.length}
+      />,
+    );
+    const instance = component.getInstance();
+
+    expect(() => instance.scrollToIndex({index: 1})).toThrow(
+      'scrollToIndex out of range: item length 0 but minimum is 1',
+    );
+  });
+
+  it('throws if using scrollToIndex when requested index is bigger than or equal to item length', () => {
```

Comment:
This one I’m not sure about. With VirtualizedList you can do tail loading. So when you scroll to the bottom of the list you can fetch more data and render it. If scrollToIndex is greater than the data length I’d expect fail loading to be possible and for it to continue trying to scroll to the index. I’m not sure what the current behavior is but I’m a bit concerned about throwing in this case. 

Can you either remove this condition or do a bit more investigation into that edge case?

Reply:
I remove this condition.